tests/test_gui_auto_offscreen.py

```diff
@@ -0,0 +1,49 @@
+"""
+Test the force offscreen auto gui mechanism.
+"""
+
+import os
+
+import wgpu.backends.rs  # noqa
+from pytest import fixture, skip
+from testutils import can_use_wgpu_lib
+
+
+if not can_use_wgpu_lib:
+    skip("Skipping tests that need the wgpu lib", allow_module_level=True)
+
+
+@fixture(autouse=True, scope="module")
+def force_offscreen():
+    os.environ["WGPU_FORCE_OFFSCREEN"] = "true"
+    try:
+        yield
+    finally:
+        del os.environ["WGPU_FORCE_OFFSCREEN"]
+
+
+def test_canvas_class():
+    """Check if we get an offscreen canvas when the WGPU_FORCE_OFFSCREEN
+    environment variable is set."""
+    from wgpu.gui.auto import WgpuCanvas
+    from wgpu.gui.offscreen import WgpuManualOffscreenCanvas
+
+    assert WgpuCanvas is WgpuManualOffscreenCanvas
+
+
+def test_event_loop():
+    """Check that the event loop handles queued tasks and then returns.
+    TODO: This test may hang indefinitely in the failure case, can we
+    prevent that?"""
```

Comment:
Maybe something like this?
```py
def stopper():
    asyncio.get_running_loop().stop()

loop = asyncio.get_event_loop()
loop.call_later(5, stopper)
```

Reply:
Then it's guaranteed to stop, but that's a property I want to test :) you see the problem here?

Reply:
Well, you could test that the loop has not stopped by the stopper, by making it set a flag and then asserting that the flag was not set, I think?

Reply:
We could use the decorator of the pytest-timeout plugin?

Reply:
Decided to just leave it for now...